Fix invoice formatting bugs

shalapatil opened the pull request:
## Notion card

## Summary
Addressed following bugs in the invoice PDF:
- the address is getting merged with the amount in next section on PDF
- The invoice issue and due date should be displayed in dd-mm-yyyy format
- In the line entry the date should be displayed in dd-mm-yyyy format
- The currency symbol should be displayed instead of USD in amount.
[invoice_c21.pdf](https://github.com/saeloun/miru-web/files/8760708/invoice_c21.pdf)


## Preview
<!-- Please include a short loom video previewing the changes made in the PR. This will help
the reviewers visualize and get context at a glance -->

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. Please also list any relevant details for your
test configuration -->

### Checklist:

- [x] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

supriya3105 commented:
@shalapatil please resolve the conflicts

rohitjoshixyz commented on app/services/invoice_payment/pdf_generation.rb:
```diff
@@ -52,5 +57,31 @@ def format_invoice(invoice_line_items)
         }
       end
+
+      def locale(base_currency)
+        case base_currency
+        when "INR"
+          "en-IN"
+        when "USD"
+          "en-US"
+        when "EUR"
+          "de-DE"
+        when "JPY"
+          "ja-JP"
+        when "GBP"
+          "en-GB"
+        when "CAD"
+          "en-CA"
+        when "AUD"
+          "en-AU"
+        else
+          "en-US"
```
I see similar logic implemented in front-end and is already merged so this looks okay to me

supriya3105 commented:
@apoorv1316 Why hasn't this been test yet ? Please test this and merge is everything is right

akhilgkrishnan commented on Gemfile:
```diff
@@ -184,4 +184,6 @@ gem "searchkick"
 # PDF generator
 gem "grover"
 
+gem "rails-i18n", "~> 7.0.0"
```
@shalapatil Why did you add this gem?

shalapatil commented:
@akhilgkrishnan Removing this gem. Initially I added this gem, because I was settings i18n localization and Money gen was not supporting en-IN.  But yesterday, I removed code for setting i18n localization. Pushing PR for removing the gem.
